sklearn/tests/test_dummy.py

```diff
@@ -101,6 +100,23 @@ def test_most_frequent_and_prior_strategy():
                                       clf.class_prior_.reshape((1, -1)) > 0.5)
 
 
+def test_most_frequent_and_prior_strategy_with_pandas_dataframe():
+    X = [[0], [0], [0], [0]]  # ignored
+    y = [[1], [2], [1], [1]]
```

Comment:
So you copied this test from the function above and just added brackets around y so that it is 2d. Maybe something simpler would be to compare the `predict` or `y_1d` and `y_2d`.